app/controllers/posts_controller.rb

```diff
@@ -13,15 +13,19 @@ def new
   end
 
   def create
```

Comment:
- [ ] [REQUIRED] Hello team, so I discovered that I need to create a Post first before signup. I have checked your `routes.rb` file but everything is fine. I can see that the ` devise_for :users` is taking precedence, however, we can fix this, you can modify the `before_action` filter in the `PostsController` to require authentication (except for the index and show actions) so that users must be **signed in** before accessing any actions related to post creation.

``` ruby
class PostsController < ApplicationController
  before_action :authenticate_user!, except: [:index, :show]

  # Rest of the code...
end
```
### Take a look at this [video](https://vimeo.com/manage/videos/848345130/0c3933502f?studio_recording=true&record_session_id=7d63a83d-d740-47b8-87c0-1a2bc686b26b) , it shows how your app works.

### Issue resolved [clip](https://vimeo.com/manage/videos/848358312/55a9de3185?studio_recording=true&record_session_id=1f0c81b9-8461-4cb5-afb2-330650ec58cc) after implementation of my suggested changes

By adding `before_action :authenticate_user!`, you ensure that users must be signed in to access any action in the PostsController, except for the index and show actions. If a user is not signed in and tries to access the new, create, or any other action except for index and show, Devise will automatically redirect them to the login page (new_user_session_path) for authentication.

- Even though, I have debugged, and do not get why the root route is taking us to the Create Post page instead of the devise signup page.

- With my suggested change, users will be required to sign up or log in first before creating a post. Once signed in, they can access the post creation page and create posts as expected.